plugins/lookup/consul_kv.py

```diff
@@ -118,25 +126,27 @@ def run(self, terms, variables=None, **kwargs):
             raise AnsibleError(
                 'python-consul is required for consul_kv lookup. see http://python-consul.readthedocs.org/en/latest/#installation')
 
+        # get options
+        self.set_options(direct=kwargs)
+
+        scheme = self.get_option('scheme')
+        host = self.get_option('host')
+        port = self.get_option('port')
+        url = self.get_option('url')
+        if url is not None:
+            u = urlparse(url)
+            scheme = u.scheme
```

Comment:
```suggestion
            if u.scheme:
                scheme = u.scheme
```
Otherwise you'll end up with an empty value if `url` did not contain a scheme.